Project sprint 4

MikhailFrantsuzov opened the pull request:
Выполнена задача спринта 4. Экраны соответствуют макетам Figma, реализована логика приложения согласно функциональным и техническим требованиям.
Дополнительно: для кнопок "Нет" и "Да" добавлено свойство isEnabled и реализован отклик на их нажатие, путём отрисовки их рамок.

MikhailFrantsuzov committed:
Добавлены шрифты, цвета, картинки

MikhailFrantsuzov committed:
Добавлена копия кнопки "Да" из Figma

MikhailFrantsuzov committed:
Выполнена задача спринта 4. Экраны соответствуют макетам Figma, реализована логика приложения согласно функциональным и техническим требованиям.
Дополнительно: для кнопок "Нет" и "Да" добавлено свойство isEnabled и реализован отклик на их нажатие, путём отрисовки их рамок.

DanilaG commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,3 +1,10 @@
 import UIKit
 
-extension UIColor { }
+extension UIColor {
+    static var ypGreen: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
+    static var ypRed: UIColor { UIColor(named: "YP Red") ?? UIColor.red }
+    static var ypBlack: UIColor { UIColor(named: "YP Black") ?? UIColor.black}
+    static var ypBackground: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
+    static var ypGray: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
+    static var ypWhite: UIColor { UIColor(named: "YP White") ?? UIColor.white}
+}
```
Классно, что цвета в отдельных константах. Это поможет упростить работу с автодополнением + снизит шансы опечаток в названии цветов

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,134 @@ import UIKit
 
 final class MovieQuizViewController: UIViewController {
     // MARK: - Lifecycle
+    struct QuizQuestion {
+        let image: String
+        let text: String
+        let correctAnswer: Bool
+    }
+    struct QuizStepViewModel {
+        let image: UIImage
+        let question: String
+        let questionNumber: String
+    }
+    struct QuizResultsViewModel {
+        let title: String
+        let text: String
+        let buttonText: String
+    }
```
🔥, что данные в отдельных структурках. Это поможет в дальнейшем)

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,134 @@ import UIKit
+        let givenAnswer = false
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+        noButton.layer.masksToBounds = true
+        noButton.layer.borderWidth = 4
+        noButton.layer.borderColor = UIColor.ypBackground.cgColor
```
Классная кастомизация)

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,134 @@ import UIKit
+        show(quiz: viewModel)
+    }
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        let questionStep = QuizStepViewModel(image: UIImage(named: model.image) ?? UIImage(), question: model.text, questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)")
```
Давай сделаем строчку по короче, разбив её на несколько) Это поможет сделать код более читаемым.

Так, например гугл [в своём кодстайле к swift](https://google.github.io/swift/#:~:text=Swift%20code%20has%20a%20column%20limit%20of%20100%20characters.), рекомендует использовать строки не более 100 символов в длину

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,134 @@ import UIKit
+    ]
+    private var currentQuestionIndex = 0
+    private var correctAnswers = 0
+    @IBOutlet weak var noButton: UIButton!
+    @IBOutlet weak var yesButton: UIButton!
```
Давай сделаем также приватными, как этого просят требования
Это ограничит пользователей твоего класса, не дав им ввести объект в невалидное состояние)

MikhailFrantsuzov committed:
1. noButton и yesButton изменены на приватные.
2. Цвет Status Bar изменён на белый.
3. Строка с private func convert разбита на несколько строк для уменьшения длины кода.

MikhailFrantsuzov commented:
1. noButton и yesButton изменены на приватные.
2. Цвет Status Bar изменён на белый.
3. Строка с private func convert разбита на несколько строк для уменьшения длины кода.
